src/app/ReadHandler.cpp

```diff
@@ -919,5 +919,17 @@ void ReadHandler::ClearStateFlag(ReadHandlerFlags aFlag)
     SetStateFlag(aFlag, false);
 }
 
+size_t ReadHandler::GetReportBufferMaxSize()
+{
+    size_t maxBufSize                  = kMaxSecureSduLengthBytes;
+    Transport::SecureSession * session = GetSession();
+    if (session && session->AllowsLargePayload())
+    {
+        maxBufSize = kMaxLargeSecureSduLengthBytes;
+    }
+
+    return maxBufSize;
```

Comment:
```suggestion
    Transport::SecureSession * session = GetSession();
    if (session && session->AllowsLargePayload())
    {
        return kMaxLargeSecureSduLengthBytes;
    }
    return kMaxSecureSduLengthBytes;
```

This is to reduce the usage and scope of a local variable. Otherwise when I see the final return, I need to go up the function to read the initial assignment.